### PR TITLE
Introduce new Value ctor for NaN, Infinity regard to fix arm clang issue

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/ValueInlines.h [hunk 1]
```diff
@@ -607,6 +607,20 @@ ALWAYS_INLINE bool Value::isInt32ConvertibleDouble(const double& d, int32_t& asI
...
+    if (tag == InfinityInitTag::PostiveInfinityInit) {
+        *this = Value(EncodeAsDouble, std::numeric_limits<double>::infinity());
+    } else {
+        *this = Value(EncodeAsDouble, -std::numeric_limits<double>::infinity());
+    }
+}
+
 inline Value::Value(const double& d)
 {
     int32_t asInt32;
```
Comment (clover2123) [inline src/runtime/ValueInlines.h:626]: It seems that `Value::Value(const double& d)` method could not process Nan or infinite value correctly for arm-clang environment. 
I wonder what the main reason of the bug was

File: src/runtime/ValueInlines.h [hunk 2]
```diff
@@ -607,6 +607,20 @@ ALWAYS_INLINE bool Value::isInt32ConvertibleDouble(const double& d, int32_t& asI
...
+    *this = Value(EncodeAsDouble, std::numeric_limits<double>::quiet_NaN());
+}
+
+inline Value::Value(InfinityInitTag tag)
+{
+    if (tag == InfinityInitTag::PostiveInfinityInit) {
+        *this = Value(EncodeAsDouble, std::numeric_limits<double>::infinity());
+    } else {
+        *this = Value(EncodeAsDouble, -std::numeric_limits<double>::infinity());
+    }
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/ValueInlines.h:621]: I see!

File: src/runtime/ValueInlines.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/ValueInlines.h:626]: arm-clang could not process well when input ctor to const inf, nan value.
I don't know why if could happen.
If we use O0~1 options, it is not happend

File: src/runtime/ValueInlines.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/ValueInlines.h]: I've updated my pr

File: src/runtime/ValueInlines.h (hunk 1 above)
Comment (clover2123) [reply src/runtime/ValueInlines.h:627]: I see. Thanks for your great work 👏

**Review by clover2123 — APPROVED** [review]

**Pull request merged**